diff --git a/Runtime/Scripts/Features/PXR_CompositionLayer.cs b/Runtime/Scripts/Features/PXR_CompositionLayer.cs
--- a/Runtime/Scripts/Features/PXR_CompositionLayer.cs
+++ b/Runtime/Scripts/Features/PXR_CompositionLayer.cs
@@ -225,7 +225,7 @@ namespace Unity.XR.PXR
             overlayEyeCamera[1] = rightCamera;
         }
 
-        private void InitializeBuffer()
+        public void InitializeBuffer()
         {
             if (!isExternalAndroidSurface && !isClones)
             {
@@ -247,8 +247,21 @@ namespace Unity.XR.PXR
             }
             else
             {
-                overlayParam.width = 1024;
-                overlayParam.height = 1024;
+                if (layerTextures[0] != null)
+                {
+                    overlayParam.width = (uint)layerTextures[0].width;
+                    overlayParam.height = (uint)layerTextures[0].height;
+                }
+                else if (layerTextures[1] != null)
+                {
+                    overlayParam.width = (uint)layerTextures[1].width;
+                    overlayParam.height = (uint)layerTextures[1].height;
+                }
+                else
+                {
+                    overlayParam.width = 1024;
+                    overlayParam.height = 1024;
+                }
             }
 
             overlayID++;
diff --git a/Runtime/Scripts/Features/PXR_OverLay.cs b/Runtime/Scripts/Features/PXR_OverLay.cs
--- a/Runtime/Scripts/Features/PXR_OverLay.cs
+++ b/Runtime/Scripts/Features/PXR_OverLay.cs
@@ -246,8 +246,21 @@ namespace Unity.XR.PXR
             }
             else
             {
-                overlayParam.width = 1024;
-                overlayParam.height = 1024;
+                if (layerTextures[0] != null)
+                {
+                    overlayParam.width = (uint)layerTextures[0].width;
+                    overlayParam.height = (uint)layerTextures[0].height;
+                }
+                else if (layerTextures[1] != null)
+                {
+                    overlayParam.width = (uint)layerTextures[1].width;
+                    overlayParam.height = (uint)layerTextures[1].height;
+                }
+                else
+                {
+                    overlayParam.width = 1024;
+                    overlayParam.height = 1024;
+                }
             }
 
             PXR_CompositionLayer.overlayID++;
